pylint/lint/pylinter.py

```diff
@@ -898,33 +911,43 @@ def _expand_files(
     def set_current_module(self, modname: str, filepath: str | None = None) -> None:
         """Set the name of the currently analyzed module and
         init statistics for it.
+
+        Save current stats before init to make sure no counters for
+        error, statement, etc are missed.
         """
         if not modname and filepath is None:
             return
         self.reporter.on_set_current_module(modname or "", filepath)
         self.current_name = modname
         self.current_file = filepath or modname
+        self.all_stats.append(self.stats)
+        self.stats = LinterStats()
         self.stats.init_single_module(modname or "")
 
         # If there is an actual filepath we might need to update the config attribute
         if filepath:
-            namespace = self._get_namespace_for_file(
-                Path(filepath), self._directory_namespaces
+            config_path, namespace = self._get_namespace_for_file(
+                Path(filepath).resolve(), self._directory_namespaces
```

Comment:
The linked report traces the regression to the call to `resolve()`. 

I wonder if we can guard it under `not path.is_absolute()`:
```py
>>> from timeit import timeit
>>> timeit('p.is_absolute()', setup='from pathlib import Path; p=Path(".")')
0.1643185840221122
>>> timeit('p.resolve()', setup='from pathlib import Path; p=Path(".")')
10.929103292000946
```

Reply:
Edit: Seems like is_absolute() will always be false as things stand, so we probably need to look higher up the stack for a place to do some sort of conversion.